From 26e0474ccc855f0cd6d9a5ba3edb19fe5f7b89b5 Mon Sep 17 00:00:00 2001
From: Bernd Kuhls <bernd.kuhls@writeme.com>
Date: Sun, 22 May 2016 09:48:57 +0200
Subject: [PATCH 01/19] tools/bneptest.c: Remove include linux/if_bridge.h to
 fix musl build

Inspired by busybox commit:
https://git.busybox.net/busybox/commit/networking/brctl.c?id=5fa6d1a632505789409a2ba6cf8e112529f9db18

The build error was found by the autobuilders of the buildroot project:
http://autobuild.buildroot.net/results/eba/ebaa0bcb9c325aa6ed0bbd6c7ec75d44befa7645/build-end.log

Signed-off-by: Bernd Kuhls <bernd.kuhls@writeme.com>
(Patch sent upstream:
 http://article.gmane.org/gmane.linux.bluez.kernel/67621)
---
 tools/bneptest.c | 12 +++++++++++-
 1 file changed, 11 insertions(+), 1 deletion(-)

diff --git a/tools/bneptest.c b/tools/bneptest.c
index a7d5815..0f4d6ed 100644
--- a/tools/bneptest.c
+++ b/tools/bneptest.c
@@ -36,7 +36,17 @@
 #include <net/if.h>
 #include <linux/sockios.h>
 #include <netinet/in.h>
-#include <linux/if_bridge.h>
+/* #include <linux/if_bridge.h>
+ * breaks on musl: we already included netinet/in.h,
+ * if we also include <linux/if_bridge.h> here, we get this:
+ * In file included from /usr/include/linux/if_bridge.h:18,
+ *                  from networking/brctl.c:67:
+ * /usr/include/linux/in6.h:32: error: redefinition of 'struct in6_addr'
+ * /usr/include/linux/in6.h:49: error: redefinition of 'struct sockaddr_in6'
+ * /usr/include/linux/in6.h:59: error: redefinition of 'struct ipv6_mreq'
+ */
+/* From <linux/if_bridge.h> */
+#define BRCTL_SET_BRIDGE_FORWARD_DELAY 8
 
 #include <glib.h>
 
-- 
2.20.1

